lib/metadata/schema-parser.js

```diff
@@ -698,29 +765,136 @@ SchemaParserV2.prototype._parseTableOrView = function (tableInfo, encoder, table
-    tableInfo.includeAllColumns = tableRow['include_all_columns'];
-    return tableInfo;
-  }
-  tableInfo.indexes = Index.fromRows(indexRows);
-  var flags = tableRow['flags'];
-  var isDense = flags.indexOf('dense') >= 0;
-  var isSuper = flags.indexOf('super') >= 0;
-  var isCompound = flags.indexOf('compound') >= 0;
-  tableInfo.isCompact = isSuper || isDense || !isCompound;
-  //remove the columns related to Thrift
-  var isStaticCompact = !isSuper && !isDense && !isCompound;
-  if(isStaticCompact) {
-    pruneStaticCompactTableColumns(tableInfo);
-  }
-  else if (isDense) {
-    pruneDenseTableColumns(tableInfo)
-  }
-  return tableInfo;
+    if (isView) {
+      tableInfo.tableName = tableRow['base_table_name'];
+      tableInfo.whereClause = tableRow['where_clause'];
+      tableInfo.includeAllColumns = tableRow['include_all_columns'];
+      return callback();
+    }
+    tableInfo.indexes = Index.fromRows(indexRows);
+    var flags = tableRow['flags'];
+    var isDense = flags.indexOf('dense') >= 0;
+    var isSuper = flags.indexOf('super') >= 0;
+    var isCompound = flags.indexOf('compound') >= 0;
+    tableInfo.isCompact = isSuper || isDense || !isCompound;
+    //remove the columns related to Thrift
+    var isStaticCompact = !isSuper && !isDense && !isCompound;
+    if(isStaticCompact) {
+      pruneStaticCompactTableColumns(tableInfo);
+    }
+    else if (isDense) {
+      pruneDenseTableColumns(tableInfo)
+    }
+    callback();
+  });
+};
+
+/** @override */
+SchemaParserV2.prototype._parseAggregate = function (row, callback) {
+  var encoder = this.cc.getEncoder();
+  var aggregate = new Aggregate();
+  aggregate.name = row['aggregate_name'];
+  aggregate.keyspaceName = row['keyspace_name'];
+  aggregate.signature = row['argument_types'] || utils.emptyArray;
+  aggregate.stateFunction = row['state_func'];
+  aggregate.finalFunction = row['final_func'];
+  aggregate.initConditionRaw = row['initcond'];
+  var self = this;
+  async.series([
+    function parseArguments(next) {
+      async.map(row['argument_types'] || utils.emptyArray, function (name, mapNext) {
+        encoder.parseTypeName(row['keyspace_name'], name, 0, null, self.udtResolver, mapNext);
+      }, function (err, result) {
+        aggregate.argumentTypes = result;
+        next(err);
+      });
+    },
+    function parseStateType(next) {
+      encoder.parseTypeName(row['keyspace_name'], row['state_type'], 0, null, self.udtResolver, function (err, type) {
+        aggregate.stateType = type;
+        next(err);
+      });
+    },
+    function parseReturnType(next) {
+      encoder.parseTypeName(row['keyspace_name'], row['return_type'], 0, null, self.udtResolver, function (err, type) {
+        aggregate.returnType = type;
+        next(err);
+      });
+    }
+  ], function (err) {
+    if (err) {
+      return callback(err);
+    }
+    try {
+      aggregate.initCondition = encoder.decode(aggregate.initConditionRaw, aggregate.stateType);
```

Comment:
Will this work?  'initcond' is now a 'text' type, so i'm assuming this will be a string now instead of bytes.  Will debug a bit and see how it works.  I'm running into a variety of aggregate bugs around initcond in cassandra itself so hard to test at the moment.